### PR TITLE
Update HOTH specification with IGCE schema

### PR DESCRIPTION
**Pull request by juliusfitzhugh-ccpo**
Add the IGCE schema to support document generation of the IGCE document.

Summary of the information that needs to be captured is below. See [ticket comment](https://ccpo.atlassian.net/browse/AT-7757?focusedCommentId=337288) for a more in depth explanation of the IGCE document and an example with filled in data. 
### Period Sheets 
There are multiple period sheets that contain information for each period. The information required: 
- a `period of performance` - an amount of time (e.g., 6 months)
- the `funding document number` - relates to the funding plan
  - is either the GTC and order number when using G-Invoicing and the 7600 A/B documents, OR the MIPR number when using a MIPR document. 
- one or more `CLIN line items` (various properties)

### Summary Sheet
Most of the summary sheet will be filled in based upon formulas connected to the period sheets. However, there are two fields that need to be filled in: 
- the `surge capabilities` (e.g., 1-50%)
- the `funding document number` - same as the period sheet

### Supporting Documentation
On the `Instructions` sheet towards the bottom it refers to supporting documentation used for estimates that can come from CSP calculators. It is not clear which format the supporting document will be in currently (e.g., PDF, CSV, link), and that this may help else where causing the removal of the `Supporting Information for IGCE` sheet in the IGCE document.

Ticket: AT-7757

### CODE REVIEW

File: hoth_api.yaml [hunk 1]
```diff
@@ -407,10 +407,77 @@ components:
...
+          type: array
+          items:
+            $ref: "#/components/schemas/PeriodEstimate"
+    PeriodEstimate:
+      description: "Information specific to a base or option period sheet in the IGCE document"
+      type: object
+      properties:
+        period:
+          description: "The period of performance for a specific period sheet."
+          $ref: "#/components/schemas/Period"
```
Comment (zachclark-ccpo) [inline hoth_api.yaml:449]: We should use:
```
    PeriodOfPerformance:
      description: "Part of DOW, represents a single Period of Performance"
      type: object
      properties:
        base_period:
          $ref: "#/components/schemas/Period"
        option_periods:
          type: array
          items:
            $ref: "#/components/schemas/Period"
        pop_start_request:
          type: boolean
        requested_pop_start_date:
          type: string
          format: date
          example: "2021-07-01"
        time_frame:
          type: string
          enum:
            - NO_LATER_THAN
            - NO_SOONER_THAN
        recurring_requirement:
          type: boolean
```

Comment (zachclark-ccpo) [reply hoth_api.yaml:449]: Specifically for 
```
 base_period:
          $ref: "#/components/schemas/Period"
        option_periods:
          type: array
          items:
            $ref: "#/components/schemas/Period"
```

Comment (zachclark-ccpo) [reply hoth_api.yaml:449]: My concern is ordering the Periods on the backend

File: hoth_api.yaml [hunk 2]
```diff
@@ -407,10 +407,77 @@ components:
...
+        idiq_clin:
+          type: string
+          enum: 
+          - "1000_CLOUD"
+          - "2000_CLOUD_SUPPORT"
+          - "3000_OTHER"
+        dow_task_number:
+          type: string
+        service:
+          description: "The service offering for the particular line item."
```
Comment (zachclark-ccpo) [inline hoth_api.yaml:469]:
```suggestion
        service_offering:
          description: "The service offering for the particular line item."  
```

**Review by zachclark-ccpo — CHANGES_REQUESTED** [review]
Small change requested 
```
service -> service_offering
```

File: hoth_api.yaml (hunk 1 above)
Comment (juliusfitzhugh-ccpo) [reply hoth_api.yaml:449]: We would be able to identify the base period using `period_type` and then for all the option periods we can use the `option_order` to order on the backend. I'd be open to using the PoP schema, but I'd be concerned the extra data might cause validation issues.

**Review by zachclark-ccpo — APPROVED** [review]

**Pull request merged**